channel/contractbackend.go

```diff
@@ -73,11 +82,24 @@ type ContractBackend struct {
 // txFinalityDepth defines in how many consecutive blocks a TX has to be
 // included to be considered final. Must be at least 1.
 func NewContractBackend(cf ContractInterface, chainID ChainID, tr Transactor, txFinalityDepth uint64) ContractBackend {
+	// Check if the shared maps are initialized, if not, initialize them.
```

Comment:
Shouldn't this be under a `SharedMutex` lock?
If it is allowed to call this function in paralell (as the `== nil` checks suggest) you would end up with a race condition when writing to the maps in line 95 and 96.